src/main/java/net/mcreator/ui/dialogs/workspace/WorkspaceGeneratorSetupDialog.java

```diff
@@ -46,6 +47,20 @@ public static void runSetup(final MCreator m, boolean showWebsite) {
 		AtomicBoolean setupOk = new AtomicBoolean(true);
 
 		Thread t = new Thread(() -> {
+			if (m.getGeneratorConfiguration().getJDKVersion() != null) {
+				ProgressDialog.ProgressUnit p0 = new ProgressDialog.ProgressUnit(
+						L10N.t("dialog.setup_workspace.step.jdk"));
+				dial.addProgress(p0);
+
+				// Download the custom JDK if needed
+				if (SDKDownloader.downloadJDK(m.getGeneratorConfiguration())) {
+					p0.ok();
+				} else {
+					p0.err();
```

Comment:
I think that if generator downloading fails, you should stop setup here and report error as without JDK, no step further will work